components/dashboards-web-component/src/common/WidgetRenderer.jsx

```diff
@@ -0,0 +1,219 @@
+
+const widgetScriptUrlPrefix = `${window.location.origin}${window.contextPath}/public/extensions/widgets`;
+const WidgetLoadingStatus = {
+    INIT: 'init',
+    FETCHING: 'fetching',
+    FETCHING_FAIL: 'fetching-fail',
+    LOADING: 'loading',
+    LOADING_FAIL: 'loading-fail',
+    LOADED: 'loaded'
+};
+
+export default class WidgetRenderer extends Component {
+
+    constructor(props) {
+        super(props);
+        this.widgetUUID = props.id;
+        let goldenLayout = props.glContainer.layoutManager;
+        let config = GoldenLayoutContentUtils.getWidgetContent(this.widgetUUID, goldenLayout.config.content);
+        this.widgetName = config.component;
+        this.widgetClass = null;
+
+        this.state = {
+            currentTheme: getMuiTheme(darkBaseTheme),
+            widgetLoadingStatus: WidgetLoadingStatus.INIT,
+            widgetFetchingProgress: -1
+        };
+        props.glEventHub.on('__mui-theme-change', (newTheme) => this.setState({currentTheme: newTheme}));
+
+        this.getWidgetClass = this.getWidgetClass.bind(this);
+        this.loadWidgetClass = this.loadWidgetClass.bind(this);
+        this.fetchWidget = this.fetchWidget.bind(this);
+        this.updateWidgetLoadingStatus = this.updateWidgetLoadingStatus.bind(this);
+        this.componentDidMount = this.componentDidMount.bind(this);
+    }
+
+    getWidgetClass() {
+        if (!this.widgetClass) {
+            this.widgetClass = WidgetClassRegistry.getWidgetClass(this.widgetName);
+        }
+        return this.widgetClass;
+    }
+
+    loadWidgetClass() {
+        if (this.getWidgetClass()) {
+            this.updateWidgetLoadingStatus(WidgetLoadingStatus.LOADED);
+        } else {
+            this.fetchWidget();
+        }
+    }
+
+    fetchWidget() {
+        this.updateWidgetLoadingStatus(WidgetLoadingStatus.FETCHING);
+        Axios.create({
+            baseURL: widgetScriptUrlPrefix,
+            timeout: 10000,
+            onDownloadProgress: (progressEvent) => {
+                let progress = progressEvent.lengthComputable ?
+                               ((progressEvent.loaded / progressEvent.total) * 100) : -1;
+                this.updateWidgetLoadingStatus(WidgetLoadingStatus.FETCHING, progress);
+            }
+        })
+            .get(`/${this.widgetName}/${this.widgetName}.js`)
+            .then(response => {
+                this.updateWidgetLoadingStatus(WidgetLoadingStatus.LOADING);
+                window.eval(response.data);
+                if (this.getWidgetClass()) {
+                    // TODO: 5/5/18 Wire pub/sub
+                    this.updateWidgetLoadingStatus(WidgetLoadingStatus.LOADED);
+                } else {
+                    this.updateWidgetLoadingStatus(WidgetLoadingStatus.FETCHING_FAIL);
+                }
+            })
+            .catch(error => {
+                console.warn(`Fetching widget '${this.widgetName}' failed.` + error.message);
```

Comment:
Shall we check for the availability of the console object as this is not supported by all browsers? ex: IE<10

Reply:
IMO, we don't need to support IE<10.
Let's change this log to debug level